staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors.go

```diff
@@ -221,6 +222,9 @@ func InternalError(field *Path, err error) *Error {
 	return &Error{ErrorTypeInternal, field.String(), nil, err.Error()}
 }
 
+func ChangeForbidden(field *Path, value interface{}, detail string) *Error {
+	return &Error{ErrorTypeChangeForbidden, field.String(), value, detail}
+}
```

Comment:
@liggitt 
- Should we set `detail string` to `CauseTypeFieldChangeForbidden CauseType = "FieldChangeForbidden"` in this function while returning the Error.
- Also could you please elaborate this statement of yours 
```
I'm envisioning a helper that attempts to extract a typed metav1.Status from the error
```
, are there any examples in the repo where we do similar to this.

Reply:
> * Should we set `detail string` to `CauseTypeFieldChangeForbidden CauseType = "FieldChangeForbidden"` in this function while returning the Error?

no, the detail string here is the human-readable explanation of what was disallowed

when field validation errors are aggregated into a single metav1.Status error response, the type of the field error is translated into a cause type here:

https://github.com/kubernetes/kubernetes/blob/c964ef8d8aca4f25333c80d30a4f8feae8db6583/staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go#L281-L302

> I am not able to understand how we will do that, are there any examples in the repo where we do similar to this.

The example I linked does that:

https://github.com/kubernetes/kubernetes/blob/c964ef8d8aca4f25333c80d30a4f8feae8db6583/staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go#L808-L810

The `status.Status()` function returns a metav1.Status for inspection

Reply:
that said, we first have to make the server consistently return the data before clients can start relying on it. I think the helper method to inspect an error is useful to make now for testing purposes, but I wouldn't expect kubectl to start using it unconditionally until multiple server versions support returning this data.